### PR TITLE
Phytosian "hair"

### PR DESCRIPTION
**Pull request by Chubbygummibear**
I could have said podperson but i like their real name
# Document the changes in your pull request
AFTER LIKE 2 MONTHS IT'S FINALLY READY (PROBABLY)

Phytosians now have 10 different hairstyles that they're free to color as they please, along with some hairstyles having flower accents that can be colored separately from the character setup screen like so
image
Only some styles have flowers, however

Cabbage and Fig
imageimage
Hibiscus and Ivy
imageimage
Orchid and Prayer
imageimage
Rose and Shrub
imageimage
Spinach and Vine
imageimage

In game:
image
image

The hair isn't actually "hair" and you can't interact with it through a mirror (maybe something for later), but it will react to radiation and other feature scrambling effects in game. I also recognize that it interacts a bit awkwardly with hats, mostly that they are large hairstyles that don't fully get covered, so I will mention this to the original sprite artist to see if they are able to do something about this
# Wiki Documentation

maybe just change the Phytosian pictures to reflect the new hairstyles or something. or just leave it, since bush heads are still a valid hair choice

# Changelog

:cl:  code by Chubbygummibear, sprites by Sheets 
rscadd: gives pod race hairstyle options
imageadd: hairstyle sprites for pods
imagedel: Phytosians no longer have brush built into their heads
/:cl:

### CODE REVIEW

**Comment by KillerOrcaCora** [comment]
Very cute sprites, Now ADD IT TO DONORRRRRRRRRRRRRRRRRRRRR!!!!!!!!!!!!!!!! ***SLAM*** ***SLAM*** ***SLAM***

**Comment by Chubbygummibear** [comment]
> Very cute sprites, Now ADD IT TO DONORRRRRRRRRRRRRRRRRRRRR!!!!!!!!!!!!!!!! _**SLAM**_ _**SLAM**_ _**SLAM**_

i'll make floor length bedhead a donor only plant hair if you become a plant main

**Comment by Sheits** [comment]
epic

**Comment by Marmio64** [comment]
i feel like having hair colors that arent the same as the rest of the plantperson makes the fake hair look too much like real hair and humanizes them more than they were already, otherwise the sprites look good.

**Comment by patpol4** [comment]
Having the hair be different color than the rest of the plang is very bad and stupid because i think it is

**Comment by Simplehorror** [comment]
I also agree that the hair colour should remain the same

**Comment by ynot01** [comment]
i hate it

it looks awesome tho

**Comment by missatessatessy** [comment]
Plants irl have different colours and flowers, it's part of what makes them pretty. I don't see why phytosians can't be as beautiful as their inspiration.
Good pr and sprites

**Review by nesisknesis — COMMENTED** [review]
Varying quality but they look good. My only gripe is that they're all very wavy and long like human hair. The only pure plant-like one is the sunflower.

**Comment by Chubbygummibear** [comment]
> Varying quality but they look good. My only gripe is that they're all very wavy and long like human hair. The only pure plant-like one is the sunflower.

I could try to lock them to being lighter colors and maybe more plantlike?, but i don't think i'll be able to get much out of the original sprite artist because they became quite busy with work and school after finishing this set unfortunately :(

**Comment by Tipy1802** [comment]
This is legally epic

**Review by nesisknesis — APPROVED** [review]
Why is it all womanly hair too

File: code/__HELPERS/mobs.dm [hunk 1]
```diff
@@ -103,7 +107,8 @@
 		"teeth" = pick(GLOB.teeth_list),
 		"dome" = pick(GLOB.dome_list),
 		"dorsal_tubes" = pick(GLOB.dorsal_tubes_list),
-		"ethereal_mark" = pick(GLOB.ethereal_mark_list)
+		"ethereal_mark" = pick(GLOB.ethereal_mark_list),
+		"pod_hair" = pick(GLOB.pod_hair_list),
```
Comment (JamieD1) [inline code/__HELPERS/mobs.dm]:
```suggestion
		"pod_hair" = pick(GLOB.pod_hair_list)
```

File: code/_globalvars/lists/flavor_misc.dm [hunk 2]
```diff
@@ -46,6 +46,9 @@ GLOBAL_LIST_EMPTY(caps_list)
 GLOBAL_LIST_EMPTY(ethereal_mark_list) //ethereal face marks
 GLOBAL_LIST_INIT(color_list_ethereal, list("F Class(Green)" = "97ee63", "F2 Class (Light Green)" = "00fa9a", "F3 Class (Dark Green)" = "37835b", "M Class (Red)" = "9c3030", "M1 Class (Purple)" = "ee82ee", "G Class (Yellow)" = "fbdf56", "O Class (Blue)" = "3399ff", "A Class (Cyan)" = "00ffff"))
 
+GLOBAL_LIST_EMPTY(pod_hair_list) //ethereal face marks
+GLOBAL_LIST_EMPTY(pod_flower_list) //ethereal face marks
+
```
Comment (JamieD1) [inline code/_globalvars/lists/flavor_misc.dm]:
```suggestion
```

File: code/modules/admin/create_mob.dm [hunk 3]
```diff
@@ -38,6 +38,9 @@
 	H.dna.features["dome"] = pick(GLOB.dome_list)
 	H.dna.features["dorsal_tubes"] = pick(GLOB.dorsal_tubes_list)
 	H.dna.features["ethereal_mark"] = pick(GLOB.ethereal_mark_list)
+	H.dna.features["pod_hair"] = pick(GLOB.pod_hair_list)
+	H.dna.features["pod_flower"] = GLOB.pod_flower_list[H.dna.features["pod_hair"]]
+
```
Comment (JamieD1) [inline code/modules/admin/create_mob.dm:43]:
```suggestion
```

File: code/modules/client/preferences.dm [hunk 4]
```diff
@@ -565,6 +565,35 @@ GLOBAL_LIST_EMPTY(preferences_datums)
...
+
+				mutant_category++
+				if(mutant_category >= MAX_MUTANT_ROWS)
+					dat += "</td>"
+					mutant_category = 0
+			
+			if("pod_flower" in pref_species.default_features)
+				if(!mutant_category)
+					dat += APPEARANCE_CATEGORY_COLUMN
+
```
Comment (JamieD1) [inline code/modules/client/preferences.dm]:
```suggestion
```

File: code/modules/client/preferences.dm [hunk 5]
```diff
@@ -565,6 +565,35 @@ GLOBAL_LIST_EMPTY(preferences_datums)
 					dat += "</td>"
 					mutant_category = 0
 
+			if("pod_hair" in pref_species.default_features)
+				if(!mutant_category)
+					dat += APPEARANCE_CATEGORY_COLUMN
+
+				dat += "<h3>Head Vegitation Style</h3>"
+
```
Comment (JamieD1) [inline code/modules/client/preferences.dm]:
```suggestion
```

File: yogstation/code/modules/mob/living/carbon/human/species_types/plantpeople.dm [hunk 6]
```diff
@@ -6,8 +6,8 @@
 	id = "pod" // We keep this at pod for compatibility reasons
 	default_color = "59CE00"
 	species_traits = list(MUTCOLORS,EYECOLOR,HAS_FLESH,HAS_BONE)
-	mutant_bodyparts = list()
-	default_features = list()
+	mutant_bodyparts = list("pod_hair", "pod_flower")
+	default_features = list("mcolor" = "0F0", "pod_hair" = "Cabbage", "pod_flower" = "Cabbage", )
```
Comment (JamieD1) [inline yogstation/code/modules/mob/living/carbon/human/species_types/plantpeople.dm]:
```suggestion
	default_features = list("mcolor" = "0F0", "pod_hair" = "Cabbage", "pod_flower" = "Cabbage")
```

**Review by JamieD1 — CHANGES_REQUESTED** [review]

**Review by JamieD1 — APPROVED** [review]

**Pull request merged**